backend/src/main/java/wooteco/retrospective/application/AttendanceService.java

```diff
@@ -0,0 +1,40 @@
+package wooteco.retrospective.application;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import wooteco.retrospective.dao.attendance.AttendanceDao;
+import wooteco.retrospective.dao.attendance.TimeDao;
+import wooteco.retrospective.dao.member.MemberDao;
+import wooteco.retrospective.domain.attendance.Attendance;
+import wooteco.retrospective.domain.attendance.Time;
+import wooteco.retrospective.domain.member.Member;
+import wooteco.retrospective.dto.AttendanceRequest;
+
+@Service
+public class AttendanceService {
+
+    private final AttendanceDao attendanceDao;
+    private final MemberDao memberDao;
+    private final TimeDao timeDao;
+
+    @Autowired
+    public AttendanceService(AttendanceDao attendanceDao, MemberDao memberDao, TimeDao timeDao) {
+        this.attendanceDao = attendanceDao;
+        this.memberDao = memberDao;
+        this.timeDao = timeDao;
+    }
+
+    public Attendance postAttendance(AttendanceRequest attendanceRequest) {
+        if(attendanceDao.isExistSameTime(attendanceRequest.getMemberId(), attendanceRequest.getTimeId())) {
+            throw new IllegalArgumentException("이미 등록된 시간입니다.");
+        }
+
+        Member member = memberDao.findById(attendanceRequest.getMemberId()).orElseThrow(RuntimeException::new);
+        Time time = timeDao.findById(attendanceRequest.getTimeId()).orElseThrow(RuntimeException::new);
+        Attendance attendance = new Attendance(attendanceRequest.getDate(), member, time);
+
+        return attendanceDao.findById(attendanceDao.insert(attendance)).orElseThrow(RuntimeException::new);
+    }
```

Comment:
```suggestion
  public Attendance postAttendance(AttendanceRequest attendanceRequest) {
        if(attendanceDao.isExistSameTime(
                attendanceRequest.getMemberId(),
                attendanceRequest.getTimeId()
        )) {
            throw new IllegalArgumentException("이미 등록된 시간입니다.");
        }

        Member member = memberDao.findById(attendanceRequest.getMemberId())
                .orElseThrow(RuntimeException::new);
        Time time = timeDao.findById(attendanceRequest.getTimeId())
                .orElseThrow(RuntimeException::new);
        Attendance attendance = new Attendance(attendanceRequest.getDate(), member, time);

        return attendanceDao.findById(attendanceDao.insert(attendance))
                .orElseThrow(RuntimeException::new);
    }
```
너무 길어요!! 길이가!! 적절한 줄바꿈이 필요해 보입니다. Attendance를 생성하는 부분도 메소드로 분리할 수 있을 것 같아요!

```
attendanceDao.findById(attendanceDao.insert(attendance))
                .orElseThrow(RuntimeException::new);
```

이 부분은 왜 존재하는건가요?? insert에서 추출되는 객체만으로도 충분한 것 같아요.  검증을 위해서라면, 프로덕션에 검증을 위한 코드를 넣는것은 바람직하지 않아보입니다.

전체적으로 read 요청이라서 큰 문제는 없어보이지만 Transactional을 붙여주는게 좋아보입니다.

Reply:
머야 이제보니가 insert가 id를 반환하네...? 보통 객체 자체를 반환하는편 입니다....!

Reply:
저도 개인적으로 객체 자체를 반환하는 편인데! 취향차이 인것 같아요!

Reply:
와 저는 줄바꿈이 더 가독성이 떨어져보이는데... 저만 그런가요..? 우선 줄 바꾸고 메서드 분리했습니다 ㅎㅎ

저는 오히려 dao에서 insert할 때 객체를 만들어내는게 어색하게 느껴져요!
insert는 테이블에 삽입의 역할이지 그 객체까지 만들어내는게 역할은 아니라고 생각합니다!
보통 객체 자체를 반환하는 문서는 어디서 읽을 수 있을까요?👀

Reply:
@dusdn1702 EJB시절부터 내려온 유구한 전통이랍니다~ JPA구현체들도 그렇게 되어 있구요!
가독성의 측면은, 소스코드는 결국 문서이기 때문에 한눈에 읽혀야 한다고 생각해요. 한줄에 100 column을 넘지 않도록 하는게 왜 컨벤션인지, stream은 왜 줄내림이 기본 컨벤션인지 생각해 보면 이런 컨벤션의 의도를 파악할 수 있을거라 생각해요!

Reply:
호오 전통은 따라야죠 ^^;;
수정할게요~